src/frontends/onnx/frontend/src/op/shape.cpp

```diff
@@ -1,17 +1,49 @@
-// Copyright (C) 2018-2024 Intel Corporation
+// Copyright (C) 2018-2023 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include "op/shape.hpp"
 
+#include "core/null_node.hpp"
+#include "openvino/core/node_vector.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/op/broadcast.hpp"
 #include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
 
 using namespace ov::op;
 
 namespace ov {
 namespace frontend {
 namespace onnx {
 namespace op {
+
+namespace set_15 {
+
+ov::OutputVector shape(const ov::frontend::onnx::Node& node) {
+    using ov::op::util::is_null;
+    ov::OutputVector inputs{node.get_ov_inputs()};
+
+    const auto& data = inputs.at(0);
+    auto shape = data.get_shape();
+
+    auto start_val = node.get_attribute_value<int64_t>("start", 0);
+    auto end_val = node.get_attribute_value<int64_t>("end", -1);
+
+    start_val = start_val < 0 ? -1 : start_val;
+    end_val = end_val < 0 ? -1 : end_val;
+
+    auto start = v0::Constant::create(ov::element::i64, ov::Shape{1}, {start_val});
+    auto end = v0::Constant::create(ov::element::i64, ov::Shape{1}, {end_val});
+    const auto& default_step = v0::Constant::create(ov::element::i32, {1}, {1});
+    ov::Output<ov::Node> steps =
+        std::make_shared<v3::Broadcast>(default_step, std::make_shared<v3::ShapeOf>(start, ov::element::i64));
```

Comment:
ShapeOf(start) looks strange for me, because it is known and const. Could you check this place?

Reply:
Agree, `default_step` defined above should be sufficient, the `steps` as `Broadcast` + `ShapeOf` are rather redundant.